### DISCUSSION TITLE
Marked and adding a toolchain

### DISCUSSION THREAD

**Discussion by tristan-f-r** [markedjs/marked#2707] [Ideas]
I wanted to contribute to the demo page and noticed that it uses raw web assets. This is perfectly fine and aids readability, but a few worries can be elevated by using a toolchain (such as vite or parcel).

Pros:
- Browser compatibility and polyfill handling (currently, it does this manually)
- General performance gains (the demo takes some time to load, but it's not very clear why it's loading. When I first opened the demo page, its speed made me have the initial assumption that *marked* was slow (e.g. it was an assembly module); not the assets it was fetching. I had the liberty to check the networking tab to figure out what the load time is caused by, but other developers may not even check).

Cons:
- Requires a plugin to _use_ marked for rendering the `.md` files, as well as the website being a module (will add extra complexity to the repository's filesystem -- turning it into a workspace.

This is mainly just for consideration, but it can significantly help with the demo page contributing.

**Comment by UziTech** [comment]
The demo page itself only has very few assets so I don't think it would help much to bundle them. Not to mention the downloading all of the resources is less than 500 kb. It might shave off a few ms but nothing noticeable even on slow connections.

The things that is the slowest is the marked worker has to downloading marked and I don't think vite or anything will fix that and still allow the flexibility to load different versions like we do.

I would be happy to be wrong, so if you want to put together a PR and it still has all of the functionality of the current demo I'm ok with a toolchain.
